File: cpp/questions/levelOrder.cpp
#include <bits/stdc++.h>
#include <vector>
#include <deque>

using namespace std;

struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        if (root == nullptr) {
            return {};
        }
        std::vector<std::vector<int>> ans;
        std::deque<TreeNode*> dq;
        dq.push_back(root);
        while (!dq.empty()) {
            int cl = dq.size();
            std::vector<int> cv;
            cv.reserve(cl);
            while (cl--) {
                TreeNode* cn = dq.front();
                dq.pop_front();
                cv.push_back(cn->val);
                if (cn->left) {
                    dq.push_back(cn->left);
                }
                if (cn->right) {
                    dq.push_back(cn->right);
                }
            }
            ans.emplace_back(cv);
        }
        return ans;
    }
};

File: python/questions/levelOrder.py
from typing import Optional, List

class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if root is None:
            return []
        ans = []
        queue = [root]        
        while len(queue):
            ql = len(queue)
            cl = []
            while ql:
                n = queue.pop(0)
                cl.append(n.val)
                if n.left:
                    queue.append(n.left)
                if n.right:
                    queue.append(n.right)
                ql -= 1
            ans.append(cl)
        return ans

            
